testsuite/MDAnalysisTests/analysis/test_encore.py

```diff
@@ -794,3 +800,31 @@ def test_dimensionality_reduction_two_different_methods(self):
                         encore.PrincipalComponentAnalysis(dims[1])])
         assert_equal(coordinates[1].shape[0], dims[1])
 
+
+class TestEncoreImportWarnings(object):
+    def setUp(self):
+        # clear cache of encore module
+        for mod in list(sys.modules):  # list as we're changing as we iterate
+            if 'encore' in mod:
+                sys.modules.pop(mod, None)
+
+    @block_import('sklearn')
```

Comment:
Those tested packages still depend on sklarn. I shortened the list of tested imported modules. The joblib library is now a dependency and always installed.